dev-packages/e2e-tests/test-applications/default-browser/tests/performance.test.ts

```diff
@@ -0,0 +1,89 @@
+import exp from 'constants';
+import { expect, test } from '@playwright/test';
+import { waitForTransaction } from '@sentry-internal/test-utils';
+
+test('captures a pageload transaction', async ({ page }) => {
+  // `pageload` transaction
+  const transactionPromise = waitForTransaction('default-browser', async transactionEvent => {
+    return !!transactionEvent?.transaction && transactionEvent.contexts?.trace?.op === 'pageload';
+  });
+
+  await page.goto(`/`);
+
+  // `waitForTransaction` hangs here because `transaction` (or `span`) event is not emitted,
+  // so the code past this line is never executed.
+  const pageLoadTransaction = await transactionPromise;
+
+  expect(pageLoadTransaction).toEqual({
```

Comment:
I thought I'd assert on quite a few fields here since we rarely do it this detailed and I'm curious if we get inconsistencies/flakes in this simple app already. We might need to adjust some checks if they get flaky but at least locally, this 100% consistent.